packages/pricing/src/services/pricing-module.ts

```diff
@@ -247,19 +266,377 @@ export default class PricingModuleService<
       count,
     ]
   }
+  @InjectManager("baseRepository_")
+  async create(
+    data: PricingTypes.CreatePriceSetDTO[],
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<PricingTypes.PriceSetDTO[]> {
+    const priceSets = await this.create_(data, sharedContext)
+
+    return this.list(
+      { id: priceSets.filter((p) => !!p).map((p) => p!.id) },
+      {
+        relations: ["rule_types", "money_amounts", "price_rules"],
+      }
+    )
+  }
 
   @InjectTransactionManager(shouldForceTransaction, "baseRepository_")
-  async create(
+  protected async create_(
     data: PricingTypes.CreatePriceSetDTO[],
     @MedusaContext() sharedContext: Context = {}
   ) {
-    const priceSets = await this.priceSetService_.create(data, sharedContext)
+    const ruleAttributes = data
+      .map((d) => d.rules?.map((r) => r.rule_attribute) ?? [])
+      .flat()
 
-    return this.baseRepository_.serialize<PricingTypes.PriceSetDTO[]>(
-      priceSets,
+    const ruleTypes = await this.ruleTypeService_.list(
       {
-        populate: true,
+        rule_attribute: ruleAttributes,
+      },
+      {},
+      sharedContext
+    )
+
+    const ruleTypeMap = ruleTypes.reduce((acc, curr) => {
+      acc.set(curr.rule_attribute, curr)
+      return acc
+    }, new Map())
+
+    const invalidRuleAttributes = ruleAttributes.filter(
+      (r) => !ruleTypeMap.has(r)
+    )
+
+    if (invalidRuleAttributes.length > 0) {
+      throw new MedusaError(
+        MedusaError.Types.NOT_FOUND,
+        `Rule types don't exist for: ${invalidRuleAttributes.join(", ")}`
+      )
+    }
+
+    const invalidMoneyAmountRule = data
+      .map(
+        (d) => d.money_amounts?.map((ma) => Object.keys(ma.rules)).flat() ?? []
+      )
+      .flat()
+      .filter((r) => !ruleTypeMap.has(r))
+
+    if (invalidMoneyAmountRule.length > 0) {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_DATA,
+        `Rule types don't exist for money amounts with rule attribute: ${invalidMoneyAmountRule.join(
+          ", "
+        )}`
+      )
+    }
+
+    const priceSets = await Promise.all(
+      data.map(async (d) => {
+        const { rules, money_amounts, ...rest } = d
+        const [priceSet] = await this.priceSetService_.create(
+          [rest],
+          sharedContext
+        )
+
+        if (rules?.length) {
+          if (!priceSet.rule_types.isInitialized) {
+            priceSet.rule_types.init()
```

Comment:
@pKorsholm this can be removed, we can create the rule types through createRuleTypes. If there is a rule here that doesn't exist, we should throw an error.

Reply:
| this can be removed, we can create the rule types through createRuleTypes
we are not creating `RuleTypes` but instead `PriceSetRuleTypes`. We are already calling the service method to create these. I think the initialization was a bit of legacy from trying to just `add` to the collection rather than create the entities. That approach didn't work as the id doesn't get set properly when trying that 😄 

|  If there is a rule here that doesn't exist, we should throw an error.
should already be handled before we touch creation

Reply:
this is not relevant for the functionality, have removed it 😄